23.md

```diff
@@ -0,0 +1,75 @@
+
+"Social" clients that deal primarily with `kind:1` notes should not be expected to implement this NIP.
+
+### Format
+
+The `.content` of these events should be a string text in Markdown syntax, including YAML front-matter for other metadata that may be necessary.
+
+### Metadata
+
+This NIP defines only `title` as the metadata. For the date of publication the event `.created_at` field should be used, and for "tags"/"hashtags" (i.e. topics about which the event might be of relevance) the `"t"` event tag should be used, as in NIP-12.
+
+### Editability
+
+These articles are meant to be editable, so they should make use of the replaceability feature of NIP-33 and include a `"d"` tag with an identifier for the article. Clients should take care to only publish and read these events from relays that implement that. If they don't do that they should also take care to hide old versions of the same article they may receive.
+
+### Linking
+
+The article may be linked to using the NIP-19 `nref` code, which should contain both the public key of the author and the article identifier (the `"d"` tag) and some relays. With that information it is possible to locate the article.
+
+### References
+
+Writing clients should implement support for parsing pasted NIP-19 `nref` identifiers and adding them automatically to the list of `.tags` of the event, replacing the actual content with a string like `#[tag_index]` in the same way as NIP-08 -- or, if the reference is in the form of a URL (for example, `[click here](nref...)`) then they should be replaced with just the tag number directly (for example, `[click here][0]`).
+
+Reader clients should parse the Markdown and replace these references with either internal links so the referenced events can be accessed directly, with NIP-21 `nostr:` links or direct links to web clients that will handle these references.
+
+The idea here is that having these tags is that reader clients can display a list of backreferences at the bottom when one article mentions another.
+
+### `kind:1` interface
+
+In case there is the desire for users or clients to share the written article on their "social" clients, a `kind:1` event must be used.
+
+Since "social" clients aren't expected to understand this NIP, these notes should contain a URL like `nostr:nref1...` specifying the `nref1` code for the article, such that it can be rendered clickable and other users can use a different client than their "social" to read the article -- and also if their client handles both kinds of posts, their client may recognize the URL and handle it internally.
+
+For clients that want to implement the ability for users to comment on these articles. `kind:1` notes should be used as the comments. They must be made in reply to another `kind:1` as above, such that other people on "social" clients can see the comments and get the context from the note above and load the article being commented on if they so desire.
```

Comment:
> They must be made in reply to another `kind:1` as above

I'm curious why replies wouldn't just reference the original `kind:30023`. Could you explain that further? It seems a little clunky to have to publish two events of different kinds to allow comments on your blog post.

Personally I think it's kind of powerful to let a `kind:1` note link to _any other event regardless of kind_ in its "e" tag. We have seen some interesting features emerge from this in scuttlebutt clients. One example is that some clients started rendering `block` messages from other users in the main feed, which gave people a chance to have a discussion about why the block was issued, which resulted in more people blocking spammers and other bad actors more quickly. Because the replies to a block took the same form as replies to a normal post it was pretty easy to write code to display them.

Reply:
I think I was just extrapolating from my experience with https://github.com/fiatjaf/nocomment, in which initially website comments would just reference a URL, but that was broken because the social clients didn't know they were supposed to look for a URL in the tags and render it, so the solution was creating a "bridge" event.

In this case however this is not necessary at all, so this was just dumb and I will just rewrite this part and make it reference the event directly.

Reply:
A problem remains, though: what event? If the content of the referenced article is changing we must come up with a way to reference that changeable thing and not just a single event (although a single event can also be referenced).

How about `["f", "pubkey:identifier"]`? I don't know, I'm not thinking straight.

Reply:
Can't you force the client to make sure `"d"` is unique for a pubkey and then reference `["r",:pk,:d]`, finding all d-tagged events from pk, then link to latest seen? or am I missing something?

Reply:
The reason is why only the second element in the tag is indexed by relays, so we need the second element to be `":pk:d"` there for a client to display "backlinks" from other events.